library/std/src/sys/unix/fd.rs

```diff
@@ -26,6 +27,35 @@ const READ_LIMIT: usize = c_int::MAX as usize - 1;
 #[cfg(not(target_os = "macos"))]
 const READ_LIMIT: usize = libc::ssize_t::MAX as usize;
 
+#[cfg(any(target_os = "linux", target_os = "macos"))]
+fn max_iov() -> usize {
+    static LIM: AtomicUsize = AtomicUsize::new(0);
+
+    let mut lim = LIM.load(Ordering::Relaxed);
+    if lim == 0 {
+        let ret = unsafe {
+            libc::sysconf(
+                #[cfg(target_os = "linux")]
+                libc::_SC_IOV_MAX,
+                #[cfg(target_os = "macos")]
+                libc::_SC_UIO_MAXIOV,
+            )
+        };
+
+        // 1024 is the default value on modern Linux systems
+        // and hopefully more useful than `c_int::MAX`.
+        lim = if ret > 0 { ret as usize } else { 1024 };
```

Comment:
Should we set `errno=0` before `sysconf` and check it again when `ret == -1`?

Reply:
Ref https://man7.org/linux/man-pages/man3/sysconf.3p.html#RETURN_VALUE

Reply:
I though about this and decided to not try to handle/propagate errors here and silently fallback to a default value instead as
* the value will not change for the lifetime of the process and I therefore want to report an error at most once to avoid repeatedly making a failing system call
* but I also would not like the first call to `FileDesc::write_vectored` to behave observably different by reporting the error from `libc::sysconf`
* thereby logging the error seems to be the only reasonable action in addition to the fallback but I did not find any infrastructure for logging within the standard library

Reply:
But while not explicitly check for -1?

Reply:
While not explicitly check for -1?

Reply:
I would suggest using 16 as the fallback value since that is the minimum guaranteed by POSIX.

Reply:
> While not explicitly check for -1?

Accepting zero is not reasonable either as calls to `write_vectored` would not make progress using that limit, hence the check for `ret > 0` instead of checking `ret == 0` and `ret < 0` separately as the reaction is the same: use the fallback value.

> I would suggest using 16 as the fallback value since that is the minimum guaranteed by POSIX.

Will do.